src/runtime/task/task_engine.cpp

```diff
@@ -288,4 +291,49 @@ void task_engine::get_queue_info(/*out*/ std::stringstream &ss)
         }
     }
 }
-} // end namespace
+
+void task_engine::register_cli_commands()
+{
+    static std::once_flag flag;
+    std::call_once(flag, [&]() {
+        _task_queue_max_length_cmd = dsn::command_manager::instance().register_command(
+            {"task.queue_max_length"},
+            "task.queue_max_length <pool_code> [queue_max_length]",
+            "get/set the max task queue length of specific thread_pool, -1 means no limit",
+            [this](const std::vector<std::string> &args) {
+                if (args.size() < 1) {
+                    return std::string("ERR: invalid arguments, task.queue_max_length <pool_code> "
+                                       "[queue_max_length]");
+                }
+
+                for (auto &it : _pools)
```

Comment:
const auto

Reply:
`{ }` is too many:

```suggestion
for { } {
                    if (!it) {
                    return error....
                    }
                    if (it->_spec.pool_code.to_string() != args[0]){
                    contune
                    }
                    
                    if ...
                    
                    if ....
                    }
                    return error...
                  
```

Reply:
+1, I suggest refactor it.

Reply:
> const auto

`it->_spec.queue_length_throttling_threshold = queue_length;`
it will be changed